src/hypothesis/searchstrategy/types.py

```diff
@@ -129,6 +135,25 @@ def from_typing_type(thing):
 for t in integer_types:
     _global_type_lookup[t] = st.integers()
 
+if hasattr(os, 'PathLike'):  # pragma: no branch
+    _global_type_lookup[os.PathLike] = \
+        st.fspaths(allow_pathlike=False).map(st._PathLike)
+
+# Also check for PY3 because there is a Python 2 pathlib port on pypi
+# which wont work
```

Comment:
It would be fine to add an extra build step which installs the module under Python 2 and only runs the file system tests FWIW (it'd probably need adding to both the `appveyor.yml` and as a separate travis job). Your choice whether you care enough to do this though.